src/buffer_storage.cc

```diff
@@ -34,8 +34,9 @@ namespace detail {
 		const auto target_base_offset = get_linear_index(target_range, target_offset)
 		                                - get_linear_index(cl::sycl::range<2>{target_range[1], target_range[2]}, {target_offset[1], target_offset[2]});
 		for(size_t i = 0; i < copy_range[0]; ++i) {
-			const auto source_ptr = static_cast<const std::byte*>(source_base_ptr) + elem_size * (source_base_offset + i * (source_range[1] * source_range[2]));
-			const auto target_ptr = static_cast<std::byte*>(target_base_ptr) + elem_size * (target_base_offset + i * (target_range[1] * target_range[2]));
+			const auto* const source_ptr =
+			    static_cast<const std::byte*>(source_base_ptr) + elem_size * (source_base_offset + i * (source_range[1] * source_range[2]));
+			auto* const target_ptr = static_cast<std::byte*>(target_base_ptr) + elem_size * (target_base_offset + i * (target_range[1] * target_range[2]));
 			memcpy_strided(source_ptr, target_ptr, elem_size, cl::sycl::range<2>{source_range[1], source_range[2]}, {source_offset[1], source_offset[2]},
```

Comment:
Consider renaming this to `memcpy_strided_host`.

Reply:
Done!